### PR TITLE
[BEAM-3565] Clean up ExecutableStage

### PR DESCRIPTION
**Pull request by bsidhom**
Addressing minor issues from #4844.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand:
   - [ ] What the pull request does
   - [ ] Why it does it
   - [ ] How it does it
   - [ ] Why this approach
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by bsidhom** [comment]
R: @tgroh

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/ExecutableStage.java [hunk 1]
```diff
@@ -81,53 +81,53 @@
...
    *       {@link #getInputPCollection()}.
    *   <li>The output {@link PCollection PCollections} in the values of {@link
    *       PTransform#getOutputsMap()} are the {@link PCollectionNode PCollections} returned by
    *       {@link #getOutputPCollections()}.
-   *   <li>The {@link FunctionSpec} contains an {@link ExecutableStagePayload} which has its input
-   *       and output PCollections set to the same values as the outer PTransform itself. It further
-   *       contains the environment set of transforms for this stage.
+   *   <li>The {@link PTransform#getSpec()} contains an {@link ExecutableStagePayload} which has its
+   *       input and output PCollections set to the same values as the outer PTransform itself. It
+   *       further contains the environment set of transforms for this stage.
```
Comment (tgroh) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/ExecutableStage.java]: "The {@link PTransform#getSpec()} contains an {@link ExecutableStagePayload} with inputs and outputs equal to the PTransform's inputs and outputs, and transforms equal to the result of {@link #getTransforms}"

Comment (bsidhom) [reply runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/ExecutableStage.java]: Done

**Comment by tgroh** [comment]
run gradle java precommit

**Comment by tgroh** [comment]
run java gradle precommit

**Pull request merged**